include/net/cloud.h

```diff
@@ -78,6 +78,7 @@ struct cloud_event {
 	union {
 		struct cloud_msg msg;
 		int err;
+		void *ptr;
```

Comment:
Perhaps we should make this field more descriptive as we aim into a generic API? Right now, an application needs to figure out what kind of association is it dealing with, based solely on the pointer value/data. My proposal is to create something like this:
```
enum cloud_association_type {
    CLOUD_ASSOCIATION_SEQUENCE,
    CLOUD_ASSOCIATION_PIN,
}

struct cloud_association_data {
    enum cloud_association_type type;
    u8_t *buf;
    size_t len;
}
```
And then replacing the vague `void *ptr;` with `struct cloud_association_data association_data;`. IMO it would be more intuitive for the application to use. 

The naming could be a subject of change, as "association" seems longish, and shortening it to "ass" doesn't sound like a good idea :D "bond_data/bond_type" perhaps?

Reply:
Yeah, good proposal, the `*ptr` felt hackish just to get it to work (and it was, to be honest). Perhaps `pair` is an ok short name?

Reply:
Sounds fine for me.

Reply:
Updated per your suggestions and rebased.